src/hooks/usePlayground.js

```diff
@@ -53,29 +59,32 @@ function reducer(state, action) {
   }
 }
 
-function usePlayground(props) {
-  let { markup, query, onChange, instanceId, rootNode } = props || {};
+const onStateChange = ({ markup, query, result }) => {
+  globalState.save({ markup, query });
+  globalState.updateTitle(result?.expression?.expression);
+};
+
+const usePlayground = () => {
+  let { markup, query } = globalState.load();
 
   if (!markup && !query) {
     markup = defaultValues.markup;
     query = defaultValues.query;
   }
 
-  const result = parser.parse({ rootNode, markup, query, cacheId: instanceId });
+  const result = parser.parse({ markup, query });
   const [state, dispatch] = useReducer(withLogging(reducer), {
-    rootNode,
     markup,
+    markupEditor: null,
     query,
     result,
   });
 
   useEffect(() => {
-    if (typeof onChange === 'function') {
-      onChange(state);
-    }
+    onStateChange(state);
```

Comment:
The `onChange` handler has been implemented in as `prop`, because an external sync will allow us to sync to multiple locations. #152 for example, is going to need this.

Handling it in this hook might look cleaner, but it removes functionality on which we (will) depend.

Reply:
Right now we have two times the same functionality in both `Embedder` & `Playground`, so that's why I merged in into `usePlayground`.

Reply:
This isn't resolved. This functionality is required for future branches to be able to work. It doesn't make sense to remove it now, only to add it back in a couple of pull-requests later.